6574 filenames

pdurbin opened the pull request:
**What this PR does / why we need it**:

Several bugs were found while investigating filenames:

- From the "edit metadata for files" page, you could give files the same name.
- Via API, it was possible to give rename files to have the same name.
- Via API, it was possible to "move" files to directory with a file that has the same name, creating a duplicate.

**Which issue(s) this PR closes**:

Closes #6574

**Special notes for your reviewer**:

Methods are in IngestUtil because that's where similar methods are found. Previous methods were focused on returning a unique filename such as README-1.md. The new methods are focused on identifying duplicates.

**Suggestions on how to test this**:

Try exercising the bugs above on develop vs. this branch. The API to use is this one, passing "label" and "directoryLabel" in the JSON: http://guides.dataverse.org/en/4.20/api/native-api.html#updating-file-metadata

**Does this PR introduce a user interface change?**:

Yes, if you try to rename a file to create a naming conflict, you will see an error like this:

<img width="1148" alt="Screen Shot 2020-05-01 at 5 10 27 PM" src="https://user-images.githubusercontent.com/21006/80979124-f6c71400-8df4-11ea-956c-e466b4cb2163.png">

**Is there a release notes update needed for this change?**:

I don't believe so.

**Additional documentation**:

None.

qqmyers commented on src/main/java/edu/harvard/iq/dataverse/ingest/IngestUtil.java:
```diff
@@ -100,6 +101,80 @@ public static String duplicateFilenameCheck(FileMetadata fileMetadata, Set<Strin
+     */
+    public static boolean conflictsWithExistingFilenames(String label, String directoryLabel, List<FileMetadata> fileMetadatas, DataFile dataFile) {
+        List<String> filePathsAndNames = new ArrayList<>();
+        for (FileMetadata fileMetadata : fileMetadatas) {
```
This loop is findDuplicateFilenames() ?

sekmiller commented on src/main/java/edu/harvard/iq/dataverse/ingest/IngestUtil.java:
```diff
@@ -100,6 +101,80 @@ public static String duplicateFilenameCheck(FileMetadata fileMetadata, Set<Strin
+     */
+    public static boolean conflictsWithExistingFilenames(String label, String directoryLabel, List<FileMetadata> fileMetadatas, DataFile dataFile) {
+        List<String> filePathsAndNames = new ArrayList<>();
+        for (FileMetadata fileMetadata : fileMetadatas) {
```
would it be possible to combine this with findDuplicateFilenames so that the UI and api are using the same method? You're passing in the filemetadata list from the dataset version which is the param for findDuplicateFilenames

pdurbin replied:
@sekmiller good idea. Fixed in 8d6a5c8e6.

pdurbin commented:
I addressed code review in 8d6a5c8e6 (reduced duplicate code). I also added the new test class to the API test suite in dc77ac19f and merged the latest from develop.
